lib/internal/Magento/Framework/Setup/Declaration/Schema/SchemaConfig.php

```diff
@@ -72,10 +72,18 @@ public function getDbConfig()
      */
     public function getDeclarationConfig()
     {
-        $schema = $this->schemaFactory->create();
-        $data = $this->readerComposite->read(FileResolverByModule::ALL_MODULES);
-        $this->declarativeSchemaBuilder->addTablesData($data['table']);
-        $schema = $this->declarativeSchemaBuilder->build($schema);
-        return $schema;
+        try{
+            $schema = $this->schemaFactory->create();
+            $data = $this->readerComposite->read(FileResolverByModule::ALL_MODULES);
+            $this->declarativeSchemaBuilder->addTablesData($data['table']);
+            $schema = $this->declarativeSchemaBuilder->build($schema);
+            return $schema;
+        }catch (\Exception $e) {
+            $writer = new \Zend\Log\Writer\Stream(BP . '/var/log/schema.log');
```

Comment:
1. This is not how this framework creates objects. There is an ObjectManager for that, use it.
Inject the existing Logger class if you want to log something. 

2. Use the current log files, don't make additional log files. People are using log aggregators for the current log files. There's no point to make a log file for each type of Exception.

Reply:
> Create an issue for whatever you're attempting to fix. See if the issue is actually an issue. Then submit the PR against that issue once it's approved to be fixed.

I have already created an issue  can you please check it https://github.com/magento/magento2/issues/27241

Reply:
Yep, saw it and linked it to the PR.